dbt/adapters/sql/connections.py

```diff
@@ -199,3 +211,46 @@ def commit(self):
         connection.transaction_open = False
 
         return connection
+
+    def _retryable_cursor_execute(self,
+                                  execute_fn: Callable,
+                                  sql: str,
+                                  bindings: Optional[Any] = None,
+                                  retryable_exceptions: Iterable[Type[Exception]] = [],
+                                  retry_limit: int = 1,
+                                  retry_timeout: Union[Callable[[int], SleepTime], SleepTime] = 1,
+                                  _attempts: int = 0,
+                                  ) -> None:
+        timeout = retry_timeout(_attempts) if callable(retry_timeout) else retry_timeout
+        if timeout < 0:
```

Comment:
This feels like java style programming, aka "look before you leap". If we remove the callable variant of the timeout, can we omit this as well? In that scenario, I think it's either our default, or the user setting, both of which can be reasonably assumed to be non-negative.

Reply:
What does removing the callable variant of the timeout mean here? We're passing in a function that the downstream user implements and I personally jive with that